Relax Vulkan requirements

gdkchan opened the pull request:
Same as #4228, but now I also fixed `MaxColorAttachmentIndex` which was wrong and was breaking a lot of games. `AttachmentIndices` also had an additional index at the end for depth-stencil which was always 0. Due to that, if a depth-stencil texture was bound, `MaxColorAttachmentIndex` would be always 0. I have removed the useless 0 index at the end which should make the value of `MaxColorAttachmentIndex` correct, as it will be the index of the last color attachment as it should be.

I recommend actually testing it this time.

marysaka commented on Ryujinx.Graphics.Vulkan/VulkanRenderer.cs:
```diff
@@ -590,7 +590,11 @@ private unsafe void PrintGpuInformation()
 
             IsAmdWindows = Vendor == Vendor.Amd && RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
             IsIntelWindows = Vendor == Vendor.Intel && RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
```
Random note here: We should be using ``OperatingSystem.IsWindows()``.

riperiperi commented:
This seems to be breaking multithreaded compilation - the background compiled pipeline is different from the one requested by the first draw. Lots of logs like this:

`00:00:09.033 |W| GUI.RenderLoop Gpu TryGetGraphicsPipeline: Background pipeline compile missed on draw - incorrect pipeline state?`

Happens most notably in clubhouse games. The visuals across games seem fine.

gdkchan commented:
> This seems to be breaking multithreaded compilation - the background compiled pipeline is different from the one requested by the first draw. Lots of logs like this:
> 
> `00:00:09.033 |W| GUI.RenderLoop Gpu TryGetGraphicsPipeline: Background pipeline compile missed on draw - incorrect pipeline state?`
> 
> Happens most notably in clubhouse games. The visuals across games seem fine.

I fixed the `ColorBlendAttachmentStateCount` on `PipelineConverter` that was using a constant value before. It's the only thing I changed that could affect this I believe. Can you check if the issue is fixed?

I do still get `Background pipeline compile missed on draw - incorrect pipeline state?` warnings on some games still, but I also get them on master for at least a few months now, so it's hard to notice new warnings of this type caused by this change.

riperiperi reviewed (approved):
You're right, Pokemon Scarlet/Violet do seem to be triggering this on Master, will need to look into that at some point. The games I was getting this error on with this PR are fixed tho, and I'm not getting any other issues.

TonyMegaPro200 commented:
hi, how could i download Ryujinx for Raspbian (Raspberry Pi 4) please

gdkchan commented:
We are not releasing any Arm64 Linux build right now, you will have to build it yourself. Also it has been a few months, I don't know what is the current state of Ryujinx on the Raspberry Pi 4.

TonyMegaPro200 commented:
And do you know were can i find information for trying to build it on the raspberry pi?

gdkchan commented:
It's the same process as building on any other platform. You can just follow the guide on the README.
